shared-bindings/audiomp3/MP3Decoder.c

```diff
@@ -172,6 +172,26 @@ STATIC mp_obj_t audiomp3_mp3file_obj_set_file(mp_obj_t self_in, mp_obj_t file) {
 }
 MP_DEFINE_CONST_FUN_OBJ_2(audiomp3_mp3file_set_file_obj, audiomp3_mp3file_obj_set_file);
 
+//|     def open_file(self, str) -> None:
+//|         """Takes in the name of a mp3 file, opens it, and replaces the old playback file."""
+//|         ...
+//|
+STATIC mp_obj_t audiomp3_mp3file_obj_open_file(mp_obj_t self_in, mp_obj_t path) {
+    audiomp3_mp3file_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    mp_obj_t file;
+    check_for_deinit(self);
+    if (!mp_obj_is_str(path)) {
+        mp_raise_TypeError(translate("file path must be provided as input"));
+    }
+    file = mp_call_function_2(MP_OBJ_FROM_PTR(&mp_builtin_open_obj), path, MP_ROM_QSTR(MP_QSTR_rb));
+    if (!mp_obj_is_type(file, &mp_type_fileio)) {
+        mp_raise_TypeError(translate("file must be a file opened in byte mode"));
+    }
```

Comment:
Can skip this check, it's guaranteed that the file was just opened in byte mode by the arguments to `mp_builtin_open_obj`.